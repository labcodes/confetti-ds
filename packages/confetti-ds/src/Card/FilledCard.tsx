import React from "react";

import { CardContext } from "./contexts";

interface FilledCardProps {
  /** Components that will be rendered in the Filled Card (CardImage, CardHeader, CardAction, etc.) */
  children: React.ReactNode;
  /** Sets Filled Card's colors (mostly background colors) based on the chosen palette.  */
  color: "white" | "mineral" | "teal" | "purple";
  /** Sets the skin of the rendered Filled Card. */
  skin: "pale" | "vivid";
  /** Sets horizontal layout. It changes the position of CardImage. */
  isHorizontal?: boolean;
  /** Reduces paddings and margins on Card layout. */
  isCompact?: boolean;
  /** Add a class name to make custom changes */
  className?: string;
}

export default function FilledCard({
  children,
  color,
  skin,
  isHorizontal = false,
  isCompact = false,
  className,
}: FilledCardProps) {
  return (
    <CardContext.Provider value={{ color, skin, cardType: "filled" }}>
      <article
        className={`lab-card lab-card--filled lab-card--${color} lab-card--${skin}
      ${isCompact ? " lab-card--compact" : ""}
      ${isHorizontal ? " lab-card--horizontal" : ""}
      ${className ? ` ${className}` : ""}
    `}
      >
        {children}
      </article>
    </CardContext.Provider>
  );
}
