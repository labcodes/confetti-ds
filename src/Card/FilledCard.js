import React from "react";
import PropTypes from "prop-types";

import { CardContext } from "./contexts";
import { filledCardChoices } from "./propTypes";

export default class FilledCard extends React.Component {
  static propTypes = {
    /** Components that will be rendered in the Filled Card (CardImage, CardHeader, CardAction, etc.) */
    children: PropTypes.node.isRequired,
    /** Sets Filled Card's colors (mostly background colors) based on the chosen palette.  */
    color: PropTypes.oneOf(filledCardChoices.color).isRequired,
    /** Sets the skin of the rendered Filled Card. */
    skin: PropTypes.oneOf(filledCardChoices.skin).isRequired,
    /** Sets horizontal layout. It changes the position of CardImage. */
    isHorizontal: PropTypes.bool,
    /** Reduces paddings and margins on Card layout. */
    isCompact: PropTypes.bool,
    /** Add a class name to make custom changes */
    className: PropTypes.string,
  };

  static defaultProps = {
    isHorizontal: false,
    isCompact: false,
    className: undefined,
  };

  render() {
    const {
      children,
      color,
      skin,
      isCompact,
      isHorizontal,
      className,
    } = this.props;

    return (
      <CardContext.Provider value={{ color, skin, cardType: "filled" }}>
        <article
          className={`lab-card lab-card--filled lab-card--${color} lab-card--${skin}
          ${isCompact ? " lab-card--compact" : ""}
          ${isHorizontal ? " lab-card--horizontal" : ""}
          ${className ? ` ${className}` : ""}
        `}
        >
          {children}
        </article>
      </CardContext.Provider>
    );
  }
}
