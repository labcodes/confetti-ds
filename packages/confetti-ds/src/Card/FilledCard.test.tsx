import React from "react";
import renderer from "react-test-renderer";

import FilledCard from "./FilledCard";

describe("FilledCard", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(
        <FilledCard color="teal" skin="pale">
          Content
        </FilledCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isCompact", async () => {
    const renderedComponent = renderer
      .create(
        <FilledCard color="teal" skin="pale" isCompact>
          Content
        </FilledCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isHorizontal", async () => {
    const renderedComponent = renderer
      .create(
        <FilledCard color="teal" skin="pale" isHorizontal>
          Content
        </FilledCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with extra className", async () => {
    const renderedComponent = renderer
      .create(
        <FilledCard color="teal" skin="pale" className="test-classname">
          Content
        </FilledCard>
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
