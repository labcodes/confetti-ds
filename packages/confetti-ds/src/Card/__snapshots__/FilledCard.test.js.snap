// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilledCard renders with base props 1`] = `
<article
  className="lab-card lab-card--filled lab-card--teal lab-card--pale
          
          
          
        "
>
  Content
</article>
`;

exports[`FilledCard renders with extra className 1`] = `
<article
  className="lab-card lab-card--filled lab-card--teal lab-card--pale
          
          
           test-classname
        "
>
  Content
</article>
`;

exports[`FilledCard renders with isCompact 1`] = `
<article
  className="lab-card lab-card--filled lab-card--teal lab-card--pale
           lab-card--compact
          
          
        "
>
  Content
</article>
`;

exports[`FilledCard renders with isHorizontal 1`] = `
<article
  className="lab-card lab-card--filled lab-card--teal lab-card--pale
          
           lab-card--horizontal
          
        "
>
  Content
</article>
`;
